Update ScalarToTensorPass to correctly track the modified bit

DrJessop opened the pull request:
Summary: Update ScalarToTensorPass to correctly track the modified bit, only run export pass if there are changes to the graph.

Also updated test_export_scalar_to_tensor_pass. to_edge already implicitly ran the pass, so the additional pass was idempotent. The updated test manually constructs a graph, checks that the pass modified the graph via graph examination and running the interpreter to check for numerical correctness.

Differential Revision: D89085647

JacobSzwejbka commented on exir/passes/scalar_to_tensor_pass.py:
```diff
@@ -7,26 +7,47 @@
+
+        if changed:
+            graph_module.recompile()
+            return super().call(graph_module)
```
Why bother calling super anymore? Wont this just walk the graph doing a bunch of nothing?

abeakkas reviewed (approved):
Approved with comments on Phabricator.

DrJessop committed:
Update ScalarToTensorPass to correctly track the modified bit (#16231)

Summary:

Update ScalarToTensorPass to correctly track the modified bit.

Also updated test_export_scalar_to_tensor_pass. to_edge already implicitly ran the pass, so the additional pass was idempotent. The updated test manually constructs a graph, checks that the pass modified the graph via graph examination and running the interpreter to check for numerical correctness.

Reviewed By: hsharma35

Differential Revision: D89085647
